DDEV-Live integration for pull, fixes #2259 (#2271)
drud/ddev: docs/users/providers/DDEV-Live.md

DDEV provides a hosting solution, [DDEV-Live](http://dash.ddev.com/), as part of the full DDEV development and deployment platform. You can use DDEV-Live with DDEV-Local as you would use any hosting provider, by pushing code to Github, exporting databases to a file and then pushing them up, etc.

Visit the [documentation pages for DDEV-Live](https://docs.ddev.com/getting-started/) to get started with Drupal, TYPO3, or other PHP projects.

### DDEV-Live Quickstart

> If you have ddev installed, and have a DDEV-Live account with an active WordPress, Drupal 7, or Drupal 8 site, you can follow this quick start guide to spin up a DDEV-Live project locally. If you don't already have a DDEV-Live account, a free short-term trial is available on signup.

Reviewer: Trying to make super clear what we're talking about with DDEV, ddev, DDEV-Live, DDEV-Local, ddev-live etc. Very open to suggestions here, but seems like it'll need to be clear throughout the docs: product name vs CLI tool.

> If you have ddev installed, and have a DDEV-Live account with an active WordPress, Drupal 7, or Drupal 8 site, you can follow this quick start guide to spin up a DDEV-Live project locally. If you don't already have a DDEV-Live account, a free short-term trial is available on signup.

Reviewer: I was going to add TYPO3 but then thought perhaps the `ddev pull` command isn't ready for that yet?

1. Authenticate with DDEV-Live.

>     a. Login to your DDEV-Live Dashboard, and [obtain the DDEV API token](https://dash.ddev.com/settings/integration) for ddev to use.

Reviewer: ```suggestion
    a. Log in to your DDEV-Live Dashboard, and [obtain the DDEV-Live API token](https://dash.ddev.com/settings/integration) for ddev to use.
```

>     b. Run `ddev auth ddev-live <YOUR TOKEN>` (This is a one-time operation, and configures ddev to work with all the sites on your account.) (You can also set the default DDEV-Live "org" with `ddev auth ddev-live --default-org=<yourorg> <token>`)

Reviewer: ```suggestion
    b. Run `ddev auth ddev-live <YOUR TOKEN>` (This is a one-time operation, which configures ddev to work with all the sites on your DDEV-Live account.) (You can also set the default DDEV-Live [organization](https://docs.ddev.com/organizations/) for ddev to work with using `ddev auth ddev-live --default-org=<yourorg> <token>`)
```

Reviewer: This threw me off a little because I'm used to the syntax for setting the default org in ddev-live `ddev-live auth --default-org ddev-demo`
Any chance those syntaxes could be reconciled at some point?

Author: Well, `ddev auth <provider>` has existed a long time, I can't help it that ddev-live didn't follow it. It is default-org on both ddev and ddev-live auth though.

> 2. Choose the DDEV-Live site you want to use with ddev.

Reviewer: I think this means "knowing the sites you have on DDEV-Live, choose which one you'd like to work with here" but at first I thought it meant "choose from a list in ddev when prompted". Could you please clarify?

Author: It doesn't yet allow you to choose from a list, but that would be a reasonable feature request. Nobody's every requested that for Pantheon.

Author: Really, you'll want to try it out.

> 3. Check out the github repo of the DDEV-Live site.

Reviewer: ```suggestion
3. Check out the GitHub repo of the DDEV-Live site.
```

4. Configure the local checkout for ddev.

    * Navigate in your terminal to your checkout of the project codebase.
    * Run `ddev config ddev-live`. When asked for the project name use the exact name of the DDEV-Live project, as shown by `ddev-live list sites`.

> 5. Run `ddev pull`. The ddev environment will spin up, download the files and database, and import the database and files into the ddev environment. You should now be able to access the project locally.

Reviewer: That's a heckuva command, can't wait to try it!
